src/sentry/integrations/slack/utils/users.py

```diff
@@ -74,3 +78,55 @@ def get_slack_data_by_user(
                 slack_data_by_user[user] = slack_info_by_email[email]
                 break
     return slack_data_by_user
+
+
+# USING SDK CLIENT
+
+
+def format_slack_data_by_user(emails_by_user, slack_info_by_email):
+    slack_data_by_user: MutableMapping[User, Mapping[str, str]] = {}
+    for user, emails in emails_by_user.items():
+        for email in emails:
+            if email in slack_info_by_email:
+                slack_data_by_user[user] = slack_info_by_email[email]
+                break
+    return slack_data_by_user
+
+
+def format_slack_info_by_email(users: dict[str, Any]):
+    return {
+        member["profile"]["email"]: {
+            "email": member["profile"]["email"],
+            "team_id": member["team_id"],
+            "slack_id": member["id"],
+        }
+        for member in users
+        if not member["deleted"] and member["profile"].get("email")
+    }
+
+
+def get_slack_data_by_user_via_sdk(
+    integration: Integration | RpcIntegration,
+    organization: Organization | RpcOrganization,
+    emails_by_user: Mapping[User, Sequence[str]],
+) -> Generator[Any, Any, Any] | None:
```

Comment:
can we have a stricter contract for the data object?

Reply:
done